src/mgmt_conn.c

```diff
@@ -1278,6 +1278,24 @@ uzfs_zinfo_rebuild_from_clone(zvol_info_t *zinfo)
 	return (uzfs_zinfo_rebuild_start_threads(&mack, zinfo, 1));
 }
 
+/*
+ * This function returns TRUE if success can be sent to target
+ */
+static boolean_t
+can_start_rebuild_return_success(zvol_info_t *zinfo, char *volname)
+{
+	zvol_status_t status = uzfs_zinfo_get_status(zinfo);
+	zvol_rebuild_status_t rstatus = uzfs_zvol_get_rebuild_status(
+	    zinfo->main_zv);
+
+	// In single replica case target will send payload size as 0
+	// Return TRUE if status is HEALTHY and rebuild_status is DONE
+	if ((strcmp(volname, "") == 0) && (status == ZVOL_STATUS_HEALTHY) &&
```

Comment:
check volname[0] == '\0' instead of strcmp.

Reply:
put a check for replication_factor == 1, and quorum == 1 to make it more assertive. Which guaranties that it will happen in this case only.

Reply:
I think having a replication_factor == 1 is not necessary but I will add `quorum == 1` check.

Reply:
Updated